src/tiled/editableimagelayer.cpp

```diff
@@ -71,6 +75,37 @@ void EditableImageLayer::setImage(ScriptImage *image, const QUrl &source)
     imageLayer()->loadFromImage(QPixmap::fromImage(image->image()), source);
 }
 
+void EditableImageLayer::setRepeatHorizontally(bool repeatHorizontally)
+{
+    if (auto doc = mapDocument()) {
+        asset()->push(new ChangeImageLayerProperties(doc,
+                                                     imageLayer(),
+                                                     transparentColor(),
+                                                     imageSource(),
+                                                     repeatHorizontally,
+                                                     repeatVertically()));
+    } else {
+        imageLayer()->setRepeatHorizontally(repeatHorizontally);
+        if (!imageSource().isEmpty())
+            imageLayer()->loadFromImage(imageSource());
```

Comment:
Why would we call `loadFromImage` again?

Reply:
Because we copy and paste too much. Fixed.